source/NVDAObjects/UIA/winConsoleUIA.py

```diff
@@ -0,0 +1,64 @@
+# NVDAObjects/UIA/winConsoleUIA.py
+# A part of NonVisual Desktop Access (NVDA)
+# This file is covered by the GNU General Public License.
+# See the file COPYING for more details.
+# Copyright (C) 2019 Bill Dengler
+
+import time
+import textInfos
+import UIAHandler
+
+from scriptHandler import script
+from winVersion import winVersion
+from . import UIATextInfo
+from ..behaviors import Terminal
+
+
+class consoleUIATextInfo(UIATextInfo):
+	def __init__(self, obj, position, _rangeObj=None):
+		super(consoleUIATextInfo, self).__init__(obj, position, _rangeObj)
+		if position == textInfos.POSITION_CARET:
+			if winVersion.build >= 18362:  # Windows 10 version 1903
+				# The UIA implementation in 1903 causes the caret to be
+				# off-by-one, so move it one position to the right
+				# to compensate.
+				self._rangeObj.MoveEndpointByUnit(
+					UIAHandler.TextPatternRangeEndpoint_Start,
+					UIAHandler.NVDAUnitsToUIAUnits['character'],
+					1
+				)
+
+
+class winConsoleUIA(Terminal):
+	_TextInfo = consoleUIATextInfo
+	_isTyping = False
+	_lastCharTime = 0
+	_TYPING_TIMEOUT = 1
+
+	def _reportNewText(self, line):
+		# Additional typed character filtering beyond that in LiveText
+		if self._isTyping and time.time() - self._lastCharTime <= self._TYPING_TIMEOUT:
+			return
+		super(winConsoleUIA, self)._reportNewText(line)
+
+	def event_textChanged(self):
```

Comment:
This is now obsolete.

Reply:
Oops, thought I deleted that last time...